src/cpp/shared/combat/army.cpp

```diff
@@ -1338,9 +1333,21 @@ void army::PowEffect(int animIdx, int a3, int a4, int a5) {
         } else {
           creature->field_3 = ANIMATION_TYPE_WINCE;
           creature->field_4 = ANIMATION_TYPE_WINCE_RETURN;
+          if(creature->creatureIdx == CREATURE_CYBER_SHADOW_ASSASSIN) { // astral dodge animations
+            if (!(creature->creature.creature_flags & RETALIATED) && !this->animatingRangedAttack) {
```

Comment:
Reusing the RETALIATED flag? Clever, except that it doesn't quite work. This should also trigger for ranged attacks.